### ISSUE TITLE
calling constructors of package protected classes don't work.

### ISSUE THREAD

**Issue by benoitj** [FlexTradeUKLtd/jfixture#28]
calling fixture.create on a package private class fails.

Could be fixed by adding "constructor.setAccessible(true)" in the GenericConstructorRelay class.

This may affect methods too.

**Comment by richkeenan** [comment]
Thanks @benoitj. Thanks for looking into a solution too, I'll delve into that today

**Comment by richkeenan** [comment]
The problem cna be fixed by using,
```
clazz.getDeclaredConstructors() // instead of getConstructors()
``` 
in the `DefaultConstructorQuery` and `setAccessible(true)` as you've mentioned in the issue.

But doing this will be breaking behavioural change which worries me. If we have a class with a package private constructor and a public factory method (which isn't unlikely, although the constructor should probably be private), this will be breaking change because the constructor will now be called rather than the factory method.

Conceptually the fix is easy enough, just change the precedence of construction to,

1. public constructors
2. public factory methods
3. package private constructors
4. package private factory methods
(don't support protected and private, I think it'd be strange to support that)

The code currently tries all the public constructors, then all the factory methods so that will need tweaking to support the new algorithm, but should be easy enough to do. I'll try to find time today to have a crack at it

**Comment by benoitj** [comment]
Precedence looks good. I agree that only package private needs support but not protected or private. 

Let me know if you want me to test it out once it is ready

**Comment by richkeenan** [comment]
I've fixed this in 2.6.0, should be on Maven Central soon once they've synced with the Nexus OSS.
